stdlib/sql/sqlite.go

```diff
@@ -0,0 +1,169 @@
+package sql
+
+import (
+	"database/sql"
+	"fmt"
+	"time"
+
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/codes"
+	"github.com/influxdata/flux/execute"
+	"github.com/influxdata/flux/internal/errors"
+	"github.com/influxdata/flux/values"
+)
+
+type SqliteRowReader struct {
+	Cursor      *sql.Rows
+	columns     []interface{}
+	columnTypes []flux.ColType
+	columnNames []string
+	NextFunc    func() bool
+	CloseFunc   func() error
+}
+
+func (m *SqliteRowReader) Next() bool {
+	if m.NextFunc != nil {
+		return m.NextFunc()
+	}
+	next := m.Cursor.Next()
+	if next {
+		columnNames, err := m.Cursor.Columns()
+		if err != nil {
+			return false
+		}
+		m.columns = make([]interface{}, len(columnNames))
+		columnPointers := make([]interface{}, len(columnNames))
+		for i := 0; i < len(columnNames); i++ {
+			columnPointers[i] = &m.columns[i]
+		}
+		if err := m.Cursor.Scan(columnPointers...); err != nil {
+			return false
+		}
+	}
+	return next
+}
+
+func (m *SqliteRowReader) GetNextRow() ([]values.Value, error) {
+	row := make([]values.Value, len(m.columns))
+	for i, col := range m.columns {
+		switch col := col.(type) {
+		case bool, int64, uint64, float64, string:
+			row[i] = values.New(col)
+		case []uint8:
+			// this allows easier testing using existing methods
+			switch m.columnTypes[i] {
+			case flux.TInt:
+				newInt, err := UInt8ToInt64(col)
+				if err != nil {
+					return nil, err
+				}
+				row[i] = values.NewInt(newInt)
+			case flux.TFloat:
+				newFloat, err := UInt8ToFloat(col)
+				if err != nil {
+					return nil, err
+				}
+				row[i] = values.NewFloat(newFloat)
+			case flux.TTime:
+				t, err := time.Parse(layout, string(col))
+				if err != nil {
+					fmt.Print(err)
+				}
+				row[i] = values.NewTime(values.ConvertTime(t))
+			default:
+				row[i] = values.NewString(string(col))
+			}
+		case time.Time:
+			row[i] = values.NewTime(values.ConvertTime(col))
+		case nil:
+			row[i] = values.NewNull(flux.SemanticType(m.columnTypes[i]))
+		default:
+			execute.PanicUnknownType(flux.TInvalid)
+		}
+	}
+	return row, nil
+}
+
+func (m *SqliteRowReader) InitColumnNames(n []string) {
+	m.columnNames = n
+}
+
+func (m *SqliteRowReader) InitColumnTypes(types []*sql.ColumnType) {
+	stringTypes := make([]flux.ColType, len(types))
+	for i := 0; i < len(types); i++ {
+		switch types[i].DatabaseTypeName() {
+		case "INT", "INTEGER", "BIGINT", "SMALLINT", "TINYINT":
+			stringTypes[i] = flux.TInt
+		case "FLOAT", "DOUBLE":
+			stringTypes[i] = flux.TFloat
+		case "DATETIME", "TIMESTAMP", "DATE":
+			stringTypes[i] = flux.TTime
+		case "TEXT":
+			stringTypes[i] = flux.TString
+		case "BOOL", "BOOLEAN":
+			stringTypes[i] = flux.TInt
```

Comment:
I understand that there is no `BOOL` in sqlite3. But isn't it preferable that when we see the data in flux it really becomes bool?

Reply:
No - there is no book type in SQLite, but people sometimes alias the column. If you store true or false in the db, it quietly converts it into an int. so any column labelled as bool, will in fact be an int. the confusion, compared to something like MySQL is that SQLite will allow you to define the column type to be anything you like - unlike MySQL which changes the column type to be tinyint- so if you try to read this into a flux.TBool, it will always fail.

Reply:
Yeah, I see that makes sense from a technical perspective. I will pass this question to the team for comments. Because as a user my expectation here might be seeing boolean values in flux, although that requires an implicit conversion.